Add .editorconfig file to ensure consistent coding style across edito…

OmarBERRABEH opened the pull request:
**[FEATURE]:** Add .editorconfig file for consistent code formatting (closes #14)

## 📄 Description

This PR adds an `.editorconfig` file to the repository to ensure consistent code formatting across different editors and IDEs. As requested in issue #14, this configuration will help maintain code style consistency across the project and improve collaboration among team members.

The `.editorconfig` file provides standardized settings for:
- 2-space indentation for web technologies (HTML, CSS, JavaScript, TypeScript)
- UTF-8 encoding and LF line endings (CRLF for Windows-specific files)
- Proper handling for different file types (markdown, config files, etc.)
- Clean code practices like trimming trailing whitespace and adding final newlines
- Maximum line length of 100 characters

This addition is particularly important for a collaborative repository like `gems` that serves as a collection of frontend development resources organized by technology. Having consistent formatting will make the code examples more readable and maintainable for all contributors.

## ✅ Checklist

- [x] I have read and followed the [Contribution Guide](https://github.com/ekino/gems/wiki/Contribution-Guide).
- [x] The `.editorconfig` file follows industry best practices for frontend development
- [x] The configuration is compatible with major editors and IDEs (VS Code, WebStorm, Sublime Text, etc.)
- [x] Settings respect the existing code style conventions of the repository
- [x] Documentation on how to use EditorConfig has been added to the PR description

## 🔗 Contextual Links (optional)

- [EditorConfig Documentation](https://editorconfig.org/)
- [Issue #14: Add .editorconfig file](https://github.com/ekino/gems/issues/14)
- [Frontend EditorConfig Best Practices](https://frontend-digest.com/using-editorconfig-to-maintain-consistent-coding-styles-between-different-editors-and-ides-5259650799c0)

## 📸 Visuals (optional)

Before/After comparison showing consistent indentation across different file types:

```diff
# Before: Inconsistent indentation across files
// JavaScript with 4 spaces
- function example() {
-     const value = true;
-     return value;
- }

// CSS with tabs
- .example {
-	display: flex;
-	justify-content: center;
- }

# After: Consistent 2-space indentation
// JavaScript with 2 spaces
+ function example() {
+   const value = true;
+   return value;
+ }

// CSS with 2 spaces
+ .example {
+   display: flex;
+   justify-content: center;
+ }

Commit:
Add .editorconfig file to ensure consistent coding style across editors (closes #14)

adrgautier commented on .editorconfig:
```diff
@@ -0,0 +1,61 @@
+indent_size = 2
+
+# JavaScript, TypeScript, JSX, TSX
+[*.{js,jsx,ts,tsx}]
```
In rare occasions we can use `.mjs` extension for ESM. Should we add it to the config ?

mibled committed:
[EVOL]: address review feedback on .editorconfig

- Add .mjs (ESM) and .cjs (CommonJS) to JavaScript extensions
- Add missing final newline
